cmd/backup/create.go

```diff
@@ -0,0 +1,254 @@
+	cmd.Flags().BoolVarP(&create.allDownstream, "all-downstream", "", create.allDownstream, "Run backup for all downstreams across namespaces")
+	cmd.Flags().BoolVar(&create.ignoreDownstream, "ignore-downstream", create.ignoreDownstream, "Do not take backups for dependent downstream resources")
+	return cmd
+}
+
+func (c *createCommand) PreRunE(_ *cobra.Command, _ []string) error {
+	c.logger = logger.NewClientLogger(c.clientConfig.Log)
+	c.namespaceSurvey = survey.NewNamespaceSurvey(c.logger)
+	c.backupCreateSurvey = survey.NewBackupCreateSurvey(c.logger)
+	return nil
+}
+
+func (c *createCommand) RunE(_ *cobra.Command, _ []string) error {
+	var err error
+	namespace, err := c.namespaceSurvey.AskToSelectNamespace(c.clientConfig)
+	if err != nil {
+		return err
+	}
+	if err := c.prepareInput(); err != nil {
+		return err
+	}
+
+	if err := c.runBackupDryRunRequest(namespace.Name); err != nil {
+		c.logger.Info(logger.ColoredNotice("Failed to run backup dry run"))
+		return err
+	}
+	if c.onlyDryRun {
+		return nil
+	}
+
+	if !c.skipConfirm {
+		proceedWithBackup, err := c.backupCreateSurvey.AskConfirmToContinue()
+		if err != nil {
+			return err
+		}
+		if !proceedWithBackup {
+			return nil
+		}
+	}
+	return c.runBackupRequest(namespace)
+}
+
+func (c *createCommand) runBackupRequest(namespace *config.Namespace) error {
+	conn, err := connectivity.NewConnectivity(c.clientConfig.Host, backupTimeout)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	backup := pb.NewBackupServiceClient(conn.GetConnection())
+
+	spinner := progressbar.NewProgressBar()
+	spinner.Start("please wait...")
+
+	backupRequest := &pb.CreateBackupRequest{
+		ProjectName:                 c.clientConfig.Project.Name,
+		NamespaceName:               namespace.Name,
+		ResourceName:                c.resourceName,
+		DatastoreName:               c.storerName,
+		Description:                 c.description,
+		AllowedDownstreamNamespaces: c.getAllowedDownstreamNamespaces(namespace.Name),
+	}
+	for _, ds := range namespace.Datastore {
+		if ds.Type == c.storerName {
+			backupRequest.Config = ds.Backup
+		}
+	}
+	backupResponse, err := backup.CreateBackup(conn.GetContext(), backupRequest)
+	spinner.Stop()
+
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			c.logger.Error(logger.ColoredError("Backup took too long, timing out"))
+		}
+		return fmt.Errorf("request failed to backup job %s: %w", backupRequest.ResourceName, err)
+	}
+	c.printBackupResponse(backupResponse)
+	return nil
+}
+
+func (c *createCommand) printBackupResponse(backupResponse *pb.CreateBackupResponse) {
+	c.logger.Info(logger.ColoredSuccess("Resource backup completed successfully:"))
+	for counter, result := range backupResponse.Urn {
+		c.logger.Info(fmt.Sprintf("%d. %s", counter+1, result))
+	}
+}
+
+func (c *createCommand) getAllowedDownstreamNamespaces(namespaceName string) []string {
```

Comment:
Simlify to a function, pass allDownstream as param.
```go
func getAllowedDownstreamNamespaces(namespaceName string, allDownstream bool) []string {
  if allDownstream {
      return []string{"*"}
  }
  return []string{namespaceName}
}
```

Reply:
updated